### DISCUSSION TITLE
Inquiry Regarding Node Editing and Dragging Functionality

### DISCUSSION THREAD

**Discussion by pinpin59** [hizzgdev/jsmind#595] [General]
Hello everyone,

I hope you're doing well. I've been using your mind mapping tool and I find the drag-and-drop feature very useful for organizing my thoughts. However, I was wondering if there's a way to disable node editing while still keeping the drag-and-drop functionality intact. I want to be able to rearrange nodes easily without editing their content. Is this possible? I would appreciate any advice you can provide.

Thank you

**Comment by hizzgdev** [comment]
Hi @pinpin59 , I understand that you want to rearrange nodes, but want to keep node content readonly.

I would like to share my thoughts on mindmap editing before introducing the solution:

- mindmap editing is not equal to node editing
- node style change, node removal, node adding are all the change of mindmap
- layout changes sometimes are equal to node changes
  - `root -> A -> B` ==> `root -> B -> A` : is it node moving or node changing?

So from the aspect of a mindmap, we didn't provide 2 different options for node-level change and mindmap level change. But never mind, here are some solutions for your requirement:

### Scenario 1
let's assume the users are not hackers, and they just use mouse to drag-and-drop. Then we just need to disable dblclick_handler, so the users can not be into the edit mode. I didn't list the option in the doc, but you can use it.

```javascript
var options = {
    container:'jsmind_container',
    editable:true,
    default_event_handle: {
        enable_dblclick_handle: false // disable mouse double click
    }
};
var jm = new jsMind(options);
```
It is not a public option, which means it may be changed in the future, so you'd better test them carefully when you upgrade jsmind in the future.

### Scenario 2
Maybe the users know some keyboard shortcuts for editing the node content, we can setup the shortcut option to disable then. The option is listed in the [doc](https://github.com/hizzgdev/jsmind/blob/master/docs/en/2.options.md).

```javascript
var options = {
    container:'jsmind_container',
    editable:true,
    default_event_handle: {
        enable_dblclick_handle: false // disable mouse double click
    },
    shortcut: {
        mapping:{
            addchild: [], // set to an empty array, and it will remove shortcuts for it.
            addbrother: [],
            editnode: [],
            delnode: []
        }
    }
};
var jm = new jsMind(options);
```

### Scenario 3
Maybe you want to disable the capability of editing node content totally, even by calling the API.
Oh stop, I don't think it's reasonable. If the users are able to call api, they will definitely be able to change option, so we can't stop them from editing node.

**Reply by pinpin59** [reply]
Thank you very much for your prompt and thorough responses to my request. I truly appreciate the effort you put into providing such valuable information. Your assistance has been incredibly helpful and has exceeded my expectations. Once again, thank you for your kindness and expertise :)
